test/scaffolding/escapesequences

```diff
@@ -9,6 +9,14 @@ a = '\\\'
 a = '\\''
 a = '\\123'
 a = '\\x23'
-a = '\\xx'
+a = '\\xx'                     -- FAIL
+a = '\\x0g'                    -- FAIL
 a = [[bar\\f\\v\\bbaz]]
 a = [[\\]]
+// {"luaVersion": "5.2"}
+a = '\\[ \\] \\' \\" \\\ \\123 \\n\\r\\t'
+a = '\\256'                    -- FAIL
+a = '\\300'                    -- FAIL
+a = '\\260'                    -- FAIL
+a = '\\.'                      -- FAIL
+a = '\\a'                      -- FAIL
```

Comment:
I've made some changes to how test case source code is escaped. Please take a look, then rebase and adjust your changes accordingly.